pkg/k8s/watchers/pod.go

```diff
@@ -276,7 +277,24 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
 		// PodIPs update
 		fallthrough
 	case labelsChanged:
-		k.redirectPolicyManager.OnUpdatePod(newK8sPod)
+		lrpNeedsReassign = true
+	}
+
+	oldPodReady := slim_corev1.ConditionUnknown
+	newPodReady := slim_corev1.ConditionUnknown
+
+	for _, cond := range oldK8sPod.Status.Conditions {
+		if cond.Type == slim_corev1.PodReady {
+			oldPodReady = cond.Status
+		}
+	}
+	for _, cond := range newK8sPod.Status.Conditions {
+		if cond.Type == slim_corev1.PodReady {
+			newPodReady = cond.Status
+		}
+	}
+	if lrpNeedsReassign || (oldPodReady != newPodReady) {
+		k.redirectPolicyManager.OnUpdatePod(newK8sPod, lrpNeedsReassign, newPodReady == slim_corev1.ConditionTrue)
```

Comment:
I'm not able to post this comment at the exact code location since it's not part of your change, but we need to check the `podCondition` in `getLocalPodsForPolicy` in manager.go.

Reply:
> looks like the pod itself doesn't get deleted, just the PodCondition changes between Ready/NotReady/etc?

yup, that's what this tries to capture here, on pod delete there should be another event that goes to `OnDeletePod`

> we need to check the podCondition in getLocalPodsForPolicy in manager.go

Makes sense, this is for 'Add' operation correct?

Reply:
Done.